shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsTest.mm

```diff
@@ -3314,4 +3315,30 @@ - (void)testFlutterTouchInterceptingViewLinksToAccessibilityContainer {
   XCTAssertEqualObjects([touchInteceptorView accessibilityContainer], container);
 }
 
+- (void)testLayerPool {
+  // Create an IOSContext and GrDirectContext.
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"foobar"];
+  [engine run];
+  XCTAssertTrue([engine iosPlatformView] != nullptr);
+  auto ios_context = [engine iosPlatformView]->GetIosContext();
+  auto gr_context = ios_context->GetMainContext();
+
+  auto pool = flutter::FlutterPlatformViewLayerPool{};
+
+  // Add layers to the pool.
+  auto layer1 = pool.GetLayer(gr_context.get(), ios_context, MTLPixelFormatBGRA8Unorm);
```

Comment:
Is there a way to check this is actually released?  Or is that not a concern? I know the `__weak`/`@autoreleasepool` pattern won't work with non-Obj-C objects.

Reply:
The pool holds these in `std::shared_ptr<FlutterPlatformViewLayer>` so they'll be cleaned up via RAII. If we templatized `FlutterPlatformViewLayerPool` we could stick in some other class and manipulate a count in the ctor/dtor that we could then test against, but that probably wouldn't give us much more benefit, since it still wouldn't show that the real `FlutterPlatformView` layer properly releases its memory -- it uses C++ smart pointers for all but the `GrDirectContext` so I trust it, but it's still not tested.

An integration test that looks at overall memory is probably the way to do something like that.